### PR TITLE
Add net.sourceforge.uqm_mods.UQM-MegaMod

### PR DESCRIPTION
**Pull request by electronstudio**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)* 
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/JHGuitarFreak/UQM-MegaMod/issues/212
- [x] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

**Comment by electronstudio** [comment]
Using `--device=all` because it's a game that needs joystick access.

**Comment by electronstudio** [comment]
The `appdata.xml` and `.desktop` files will be upstreamed for next release, but this release has already been made without them, which is why they are needed here.

File: net.sourceforge.uqm_mods.UQM-MegaMod.json [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    "sources" : [
+      {
+        "type": "git",
+        "tag": "0.8.2",
+        "commit": "8a16e3abefcfe1b9859e50605b09106bcef4a3b4",
+        "url": "https://github.com/JHGuitarFreak/UQM-MegaMod.git"
+      },
+      {
+        "type": "file",
+        "url": "https://netix.dl.sourceforge.net/project/uqm-mods/MegaMod/0.8.2/Content/mm-0.8.2-content.uqm",
```
Comment (hfiguiere) [inline net.sourceforge.uqm_mods.UQM-MegaMod.json:34]:
```suggestion
        "url": "https://download.sourceforge.net/project/uqm-mods/MegaMod/0.8.2/Content/mm-0.8.2-content.uqm",
```

File: net.sourceforge.uqm_mods.UQM-MegaMod.json [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        "dest-filename": "mm-0.8.2-content.uqm",
+        "sha512": "3d995b9bc258313b933c0caab29faaf0d8e235a257dfb0d6d2574dc402ca6a0d442d2339d2664e85088555d1233750c650b0df2ed40f2f7cfde1a6f99de1ae1c"
+      },
+      {
+        "type": "file",
+        "url": "https://netix.dl.sourceforge.net/project/uqm-mods/MegaMod/0.8.2/Content/mm-0.8.2-3dovoice.uqm",
```
Comment (hfiguiere) [inline net.sourceforge.uqm_mods.UQM-MegaMod.json:40]:
```suggestion
        "url": "https://download.sourceforge.net/project/uqm-mods/MegaMod/0.8.2/Content/mm-0.8.2-3dovoice.uqm",
```

File: net.sourceforge.uqm_mods.UQM-MegaMod.json [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        "dest-filename": "mm-0.8.2-3dovoice.uqm",
+        "sha512": "a786014fa8d231e2368bb3de595da2cb2086a3fed0fbedd2ec960310ff7fdf40e0374cd453131c9e0d9c1a362ad21b143b2dee10afeceed7aa68d6d2b1f65cd3"
+      },
+      {
+        "type": "file",
+        "url": "https://netix.dl.sourceforge.net/project/uqm-mods/MegaMod/0.8.2/Content/mm-0.8.2-hd.uqm",
```
Comment (hfiguiere) [inline net.sourceforge.uqm_mods.UQM-MegaMod.json:46]:
```suggestion
        "url": "https://download.sourceforge.net/project/uqm-mods/MegaMod/0.8.2/Content/mm-0.8.2-hd.uqm",
```

File: net.sourceforge.uqm_mods.UQM-MegaMod.json [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        "dest-filename": "mm-0.8.2-hd.uqm",
+        "sha512": "f270ffeb5f103e939b121d7a96a02d8b93906e08b7534b5e01d96a884f0eecf801212f8e0f566889d00a9c783be8eb08a120bdcecf042e1a1c46e8ea636971c3"
+      },
+      {
+        "type": "file",
+        "url": "https://netix.dl.sourceforge.net/project/uqm-mods/MegaMod/0.8.2/Content/uqm-0.8.0-3DOMusicRemastered.uqm",
```
Comment (hfiguiere) [inline net.sourceforge.uqm_mods.UQM-MegaMod.json:52]:
```suggestion
        "url": "https://download.sourceforge.net/project/uqm-mods/MegaMod/0.8.2/Content/uqm-0.8.0-3DOMusicRemastered.uqm",
```

File: net.sourceforge.uqm_mods.UQM-MegaMod.json [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        "dest-filename": "uqm-0.8.0-3DOMusicRemastered.uqm",
+        "sha512": "73d5dedfeab9b86a28166ff346df684e6138fe8c7f621b7c530e6bef67c0194b7e7c505793ea8066551693b787003bd2936deff811894649a5737ceb58f48a7e"
+      },
+      {
+        "type": "file",
+        "url": "https://netix.dl.sourceforge.net/project/uqm-mods/MegaMod/0.8.2/Content/uqm-0.8.0-3dovideo.uqm",
```
Comment (hfiguiere) [inline net.sourceforge.uqm_mods.UQM-MegaMod.json:58]:
```suggestion
        "url": "https://download.sourceforge.net/project/uqm-mods/MegaMod/0.8.2/Content/uqm-0.8.0-3dovideo.uqm",
```

File: net.sourceforge.uqm_mods.UQM-MegaMod.json [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+        "type": "file",
+        "path": "net.sourceforge.uqm_mods.UQM-MegaMod.desktop"
+      },
+      {
+        "type": "file",
+        "path": "net.sourceforge.uqm_mods.UQM-MegaMod.appdata.xml"
+      }
+    ]
+  } ],
+  "finish-args" : [
```
Comment (hfiguiere) [inline net.sourceforge.uqm_mods.UQM-MegaMod.json:72]: this usually before `modules`.

File: net.sourceforge.uqm_mods.UQM-MegaMod.json [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+    "--socket=x11",
+    "--share=ipc",
+    "--device=all",
+    "--socket=pulseaudio",
+    "--share=network",
+    "--filesystem=home/.uqm-megamod:create"
```
Comment (hfiguiere) [inline net.sourceforge.uqm_mods.UQM-MegaMod.json:78]: this should be `--persist` instead.

**Comment by hfiguiere** [comment]
bot, build net.sourceforge.uqm_mods.UQM-MegaMod

**Comment by electronstudio** [comment]
bot, build net.sourceforge.uqm_mods.UQM-MegaMod

**Comment by electronstudio** [comment]
Build bot says "appstream-missing-screenshots" but I haven't changed the screenshots or any of the metadata since the previous successful build.  Try it again?

**Comment by electronstudio** [comment]
bot, build net.sourceforge.uqm_mods.UQM-MegaMod

**Comment by hfiguiere** [comment]
sourceforget is having hiccups:

```
$ wget https://uqm-mods.sourceforge.net/Images/MegaMod/31%20The%20Melnorme.png
--2024-01-19 22:00:06--  https://uqm-mods.sourceforge.net/Images/MegaMod/31%20The%20Melnorme.png
Resolving uqm-mods.sourceforge.net (uqm-mods.sourceforge.net)... 3.161.213.91, 3.161.213.87, 3.161.213.53, ...
Connecting to uqm-mods.sourceforge.net (uqm-mods.sourceforge.net)|3.161.213.91|:443... connected.
GnuTLS: A TLS fatal alert has been received.
GnuTLS: received alert [40]: Handshake failed
Unable to establish SSL connection.
```

**Comment by hfiguiere** [comment]
so it can't download the screenshots.

**Comment by electronstudio** [comment]
bot, build net.sourceforge.uqm_mods.UQM-MegaMod

**Comment by electronstudio** [comment]
Apparently they were down for maintenance.  Test build works.

File: net.sourceforge.uqm_mods.UQM-MegaMod.appdata.xml [hunk 8]
```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+    <id>net.sourceforge.uqm_mods.UQM-MegaMod</id>
+    <name>The Ur-Quan Masters HD MegaMod</name>
+    <launchable type="desktop-id">net.sourceforge.uqm_mods.UQM-MegaMod.desktop</launchable>
+    <summary>A mod for The Ur-Quan Masters</summary>
```
Comment (bbhtt) [inline net.sourceforge.uqm_mods.UQM-MegaMod.appdata.xml:6]:
```suggestion
    <summary>Mod for The Ur-Quan Masters</summary>
```

Comment (bbhtt) [reply net.sourceforge.uqm_mods.UQM-MegaMod.appdata.xml:6]: https://docs.flathub.org/docs/for-app-authors/appdata-guidelines/quality-guidelines#dont-start-with-an-article

**Comment by hfiguiere** [comment]
bot, build net.sourceforge.uqm_mods.UQM-MegaMod

**Comment by barthalion** [comment]
bot, build net.sourceforge.uqm_mods.UQM-MegaMod

**Comment by JHGuitarFreak** [comment]
Could you resend the invite?

**Comment by bbhtt** [comment]
> Thanks, could you please add push access for user @JHGuitarFreak to the repo https://github.com/flathub/net.sourceforge.uqm_mods.UQM-MegaMod ?

> Could you resend the invite?

Sent again.

**Pull request closed without merging**